### ISSUE TITLE
Loadbalancer can't see external service if hostname was used

### ISSUE THREAD

**Issue by kapolos** [rancher/rancher#7365]
**Rancher Versions:**
Server: 1.3.0
healthcheck: 0.2.0
ipsec: 0.8.1
network-services: 0.4.0
scheduler: 0.5.0

**Docker Version:**
1.12.5
**OS and where are the hosts located? (cloud, bare metal, etc):**
Centos 7, Esxi
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
Add an external service with a hostname instead of an ip
Try to add it as a target for a load balancer
**Results:** 
External service does not appear on the drop down list
**Expected:**
External service should appear on the drop-down list

**This is a regression, it was working fine with the previous load balancer version**. Current version: 0.4.9. Note that this only happens if the hostname option has been used. If an IP is used, the external service appears on the dropdown.

**Comment by vincent99** [comment]
This has never worked out been sorted and is not a regression. Pointing to a hostname properly requires a policy and DNS server to decide where to look it up and how long is allowed to be cached, etc.  Previous versions of the UI displayed the services grayed out with the label "(can't balance to hostnames)"

**Comment by kapolos** [comment]
@vincent99 I was pointing it to `rancher-agent.rancher.internal`. This way, I had a load balancer doing the SSL termination and I was connecting to the rancher server like `https://mydomain/`. This worked fine until I updated the infrastructure & loadbalancer images.

**Comment by kapolos** [comment]
For completeness, adding an external service pointing to `172.17.0.2` made my setup work again.

**Comment by DaveOHenry** [comment]
Ran into the same issue: We use an external http service for some of our workflows. We have to do some http header and path manipulation on the request as well as the response so we cannot use the external service directly.

As a workaround I created a "dummy" service that points to a non-existent ip address and a loadbalancer for this service. In the rancher gui I defined a custom backend name and in the custom haproxy.cfg a "server" line  like:
```
backend dummy_backend
server my-external-service my.external.service:80 check inter 60s rise 1 fall 3 resolvers rancher
```
I left out all of our custom http manipulation stuff in the example and changed the names.
This works, but the generated haproxy config also contains the dummy service.

To get rid of the dummy server in the haproxy config I just changed the dummy service in rancher from "ip address" to "hostname". Feels like a hack, but seems to work.
